fhir-server/src/main/java/com/ibm/fhir/server/util/IssueTypeToHttpStatusMapper.java

```diff
@@ -54,16 +54,16 @@
                             FHIRUtil.getExtensionStringValue(code, EXTENSION_URL_HTTP_FAILED_PRECONDITION) != null) {
                         return Status.PRECONDITION_FAILED;
                     } else if (issueType == IssueType.ValueSet.NOT_SUPPORTED &&
-                            "interaction".equals(FHIRUtil.getExtensionStringValue(code, EXTENSION_URL_NOT_SUPPORTED_DETAIL))) {
-                        return Status.BAD_REQUEST;
+                            "resource".equals(FHIRUtil.getExtensionStringValue(code, EXTENSION_URL_NOT_SUPPORTED_DETAIL))) {
+                        return Status.NOT_FOUND;
```

Comment:
double checking is 404 correct here? 400 being bad request...

Reply:
this class maps IssueTypes to HTTP statuses, but a could IssueTypes cannot be mapped cleanly.  so I invented this approach of adding an extension to the IssueType which carries the extra information needed to map it into a proper HTTPStatus.
in this case, the extra detail is about "what isn't supported" for the NOT_SUPPORTED value.  it isn't a perfect semantic fit with 404, but in practice we only set this `detail` extension to "resource" in cases where we receive a request to a `[base]/ResourceType` endpoint for a resource type that isn't supported...hence the mapping to 404 instead of 400 for this case.